src/test/java/com/google/api/pathtemplate/PathTemplateTest.java

```diff
@@ -487,9 +487,6 @@ public void complexResourceBasicInvalidIds() {
   @Test
   public void complexResourceMultipleDelimiters() {
     thrown.expect(ValidationException.class);
-    PathTemplate.create("projects/*/zones/.-~{zone_a}");
```

Comment:
It feels this still should be an invalid input as this is the third type of case (not mentioined in PR description and in the comment in `isSegmentBeginOrEndInvalid` method). This a 2+ consecutive delimiter characters at the beginning of a segment. I understand that it is a super unlikely scenario, but please strongly consider taking this case into account as invalid, or clarify why this still shoudl be considered valid in the PR description and `isSegmentBeginOrEndInvalid` comment)

Reply:
Even though super unlikely, but I would consider this a literal if it happens. 2+ consecutive delimiters are only a problem if they are in between `}` and '{', I can add more comments to clarify this scenario.